libfs/fs.go

```diff
@@ -276,7 +311,42 @@ func (fs *FS) ReadDir(p string) (fis []os.FileInfo, err error) {
 }
 
 // MkdirAll implements the billy.Filesystem interface for FS.
-func (fs *FS) MkdirAll(filename string, perm os.FileMode) error {
+func (fs *FS) MkdirAll(filename string, perm os.FileMode) (err error) {
+	fs.log.CDebugf(fs.ctx, "MkdirAll %s", filename)
+	defer func() {
+		fs.deferLog.CDebugf(fs.ctx, "MkdirAll done: %+v", err)
+	}()
+
+	parts := strings.Split(filename, "/")
```

Comment:
Oh there's a `Filesystem.Join` [method](https://godoc.org/github.com/src-d/go-billy#Basic) so I guess that's up to us. But the comment on `Join` seems to suggest we should treat Windows and Unix differently.